src/test/java/bio/terra/pact/provider/DataRepoProviderTest.java

```diff
@@ -0,0 +1,94 @@
+package bio.terra.pact.provider;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+import au.com.dius.pact.provider.junit5.PactVerificationContext;
+import au.com.dius.pact.provider.junitsupport.Provider;
+import au.com.dius.pact.provider.junitsupport.State;
+import au.com.dius.pact.provider.junitsupport.loader.PactBroker;
+import au.com.dius.pact.provider.spring.junit5.MockMvcTestTarget;
+import au.com.dius.pact.provider.spring.junit5.PactVerificationSpringProvider;
+import bio.terra.app.controller.GlobalExceptionHandler;
+import bio.terra.app.controller.SnapshotsApiController;
+import bio.terra.common.category.Pact;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
+import bio.terra.model.SnapshotModel;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.auth.iam.exception.IamForbiddenException;
+import bio.terra.service.dataset.AssetModelValidator;
+import bio.terra.service.dataset.IngestRequestValidator;
+import bio.terra.service.filedata.FileService;
+import bio.terra.service.job.JobService;
+import bio.terra.service.snapshot.SnapshotRequestValidator;
+import bio.terra.service.snapshot.SnapshotService;
+import bio.terra.service.snapshot.exception.SnapshotNotFoundException;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+
+@Tag(Pact.TAG)
+@Provider(Pact.PACTICIPANT)
+@PactBroker
+@ContextConfiguration(classes = {SnapshotsApiController.class, GlobalExceptionHandler.class})
+@WebMvcTest
+public class DataRepoProviderTest {
+
+  @Autowired private MockMvc mvc;
+
+  @MockBean private JobService jobService;
+  @MockBean private SnapshotRequestValidator snapshotRequestValidator;
+  @MockBean private SnapshotService snapshotService;
+  @MockBean private IamService iamService;
+  @MockBean private IngestRequestValidator ingestRequestValidator;
+  @MockBean private FileService fileService;
+  @MockBean private AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @MockBean private AssetModelValidator assetModelValidator;
+
+  private static final UUID SNAPSHOT_ID = UUID.fromString("12345678-abc9-012d-3456-e7fab89cd01e");
```

Comment:
I have yet to learn what the proper procedure is for these dummy variables; here I have defined this on the consumer side and we're repeating the same UUID here, although its exact value doesn't matter.  Is this correct or is there a way to not have to hardcode some specific but unimportant value here?

Reply:
That's a good question!

After reading https://broadworkbench.atlassian.net/wiki/spaces/IRT/pages/2787672065/Demystifying+Provider+States+Callbacks+-+pathFromProviderState and looking at our respective provider and consumer PRs, I am not convinced that matching dummy IDs in this case across provider and consumer is valuable.

WDS as consumer is already defining the desired provider state to work from, so that's what drives the behavior rather than a match of a specific snapshot ID.

I also looked at the Sam provider code and see that their provider states mock off of any IDs rather than specific ones: https://github.com/broadinstitute/sam/blob/5a91c3f81ce8941bb915745bf845782cf1e7d270/pact4s/src/test/scala/org/broadinstitute/dsde/workbench/sam/provider/SamProviderSpec.scala#L150

So I'll change the matchers here from `eq(SNAPSHOT_ID)` to `any(UUID.class)`.  Let me know if that works for you!

Reply:
Actually, I'll flip flop here -- looking at Terra Billing Profile Manager, seems like the hardcoded IDs may be preferable, but I'm still not fully wrapping my head around it, or the best way to share that state between consumers and providers. https://github.com/DataBiosphere/terra-billing-profile-manager/blob/main/service/src/test/java/bio/terra/profile/pact/provider/ProviderStateData.java

Reply:
Wow I did not get any notification that you had responded here!  What about the fancy method Josh is using in [these](https://github.com/DataBiosphere/terra-workspace-data-service/pull/372/files) [PRs](https://github.com/DataBiosphere/terra-workspace-manager/pull/1511/files#diff-1363f62b4473f466ae2a9b8865509d5383a815fa61af487b1aa095f7b5ca9031)?

Reply:
@calypsomatic great find, and @jladieu nice work!

I think this should work for us too.  It looks like what we were missing previously was setting up the provider states to accept parameters, like so:

```
  public void authenticatedAsAccount(Map<?, ?> parameters) throws InterruptedException {
    var authenticatedEmail = parameters.get("email").toString();
 ```

I can preemptively make the changes to TDR's snapshot provider states to expect a parameter map with an entry for `id`, and remove our hardcoded, needs-to-match-yours UUID.

Reply:
Done!